test/distributed/test_c10d_nccl.py

```diff
@@ -3893,20 +3900,27 @@ def test_send_recv_subgroup(self, async_op, group_rank):
 
     @requires_nccl()
     @skip_if_lt_x_gpu(4)
-    def test_broadcast_subgroup(self):
+    @parametrize("group_rank", [True, False])
+    def test_broadcast_subgroup(self, group_rank):
         world_size = 4
         if self.rank >= world_size:
             return
         subgroup = self._init_two_pg2_subgroups(world_size)
         device = torch.device("cuda:%d" % self.rank)
         if self.rank == 0 or self.rank == 2:
             x = torch.empty((10,), device=device)
-            c10d.broadcast(x, src=self.rank + 1, group=subgroup)
+            if group_rank:
+                c10d.broadcast(x, group_src=1, group=subgroup)
+            else:
+                c10d.broadcast(x, src=self.rank + 1, group=subgroup)
```

Comment:
is it? I saw a 'world_size=4' hardcode and a `if self.rank >= world_size: return` so i thought it was correct even though it insists on using exactly 4 gpus rather than being flexible to all available gpus